### PR TITLE
Add Int64Index, Float64Index, DatetimeIndex.

### PR DESCRIPTION
**Pull request by ueshin**
Adds `Int64Index`, `Float64Index`, and `DatetimeIndex` as a placeholder.
We should still add specific attributes and methods in the following PRs.

Before:

```py
>>> kdf = ks.DataFrame([1,2,3])
>>> type(kdf.index)
<class 'databricks.koalas.indexes.Index'>
```

After:

```py
>>> type(kdf.index)
<class 'databricks.koalas.indexes.numeric.Int64Index'>
```

### CODE REVIEW

File: databricks/koalas/indexes/numeric.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    Index : The base pandas Index type.
+
+    Notes
+    -----
+    An Index instance can **only** contain hashable objects.
+    """
+
+    pass
+
+
```
Comment (xinrong-databricks) [inline databricks/koalas/indexes/numeric.py:54]: May I ask for the reason why we don't support UInt64Index as https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.UInt64Index.html?

File: databricks/koalas/internal.py [hunk 2]
```diff
@@ -708,25 +708,37 @@ def offset(id):
             (sdf[offset_column] + sdf[row_number_column] - 1).alias(column_name), *scols
         )
 
-    def spark_column_name_for(self, label: Tuple) -> str:
-        """ Return the actual Spark column name for the given column label. """
-        return self.spark_frame.select(self.spark_column_for(label)).columns[0]
-
-    def spark_column_for(self, label: Tuple):
+    def spark_column_for(self, label: Tuple) -> spark.Column:
```
Comment (xinrong-databricks) [inline databricks/koalas/internal.py:711]: Nice refactoring!

**Comment by xinrong-databricks** [comment]
It might be helpful we add some code examples in the PR description to explain what changes are introduced by this PR, for example,

Before:
```
>>> kdf = ks.DataFrame([1,2,3])
>>> type(kdf.index)
<class 'databricks.koalas.indexes.Index'>
```

After:
```
>>> type(kdf.index)
<class 'databricks.koalas.indexes.numeric.Int64Index'>
```

**Comment by xinrong-databricks** [comment]
LGTM! Thank you for doing this!

File: databricks/koalas/indexes/numeric.py (hunk 1 above)
Comment (ueshin) [reply databricks/koalas/indexes/numeric.py:54]: Because Koalas doesn't support unsigned types.

**Review by HyukjinKwon — APPROVED** [review]

File: databricks/koalas/indexes/numeric.py (hunk 1 above)
Comment (xinrong-databricks) [reply databricks/koalas/indexes/numeric.py:54]: Got it! Thanks!

**Pull request merged**